string state_name                     # action to be taken
int16 state_id                 # name of the model to take action
string action_name           #
string outcome               #
bool succeed                  #
